workbench-core/authorization/src/dynamicAuthorization/ddbDynamicAuthorizationPermissionsPlugin.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  CreateIdentityPermissionsRequest,
+  CreateIdentityPermissionsResponse
+} from './dynamicAuthorizationInputs/createIdentityPermissions';
+import {
+  DeleteIdentityPermissionsRequest,
+  DeleteIdentityPermissionsResponse
+} from './dynamicAuthorizationInputs/deleteIdentityPermissions';
+import {
+  GetDynamicOperationsByRouteRequest,
+  GetDynamicOperationsByRouteResponse
+} from './dynamicAuthorizationInputs/getDynamicOperationsByRoute';
+import {
+  GetIdentityPermissionsByIdentityRequest,
+  GetIdentityPermissionsByIdentityResponse
+} from './dynamicAuthorizationInputs/getIdentityPermissionsByIdentity';
+import {
+  GetIdentityPermissionsBySubjectRequest,
+  GetIdentityPermissionsBySubjectResponse
+} from './dynamicAuthorizationInputs/getIdentityPermissionsBySubject';
+import { IdentityPermission, IdentityType } from './dynamicAuthorizationInputs/identityPermission';
+import { IsRouteIgnoredRequest, IsRouteIgnoredResponse } from './dynamicAuthorizationInputs/isRouteIgnored';
+import {
+  IsRouteProtectedRequest,
+  IsRouteProtectedResponse
+} from './dynamicAuthorizationInputs/isRouteProtected';
+import { DynamicAuthorizationPermissionsPlugin } from './dynamicAuthorizationPermissionsPlugin';
+
+export class DDBDynamicAuthorizationPermissionsPlugin implements DynamicAuthorizationPermissionsPlugin {
+  private _dynamoDBService: DynamoDBService;
+
+  public constructor(config: { dynamoDBService: DynamoDBService }) {
+    this._dynamoDBService = config.dynamoDBService;
+  }
+
+  public async isRouteIgnored(isRouteIgnoredRequest: IsRouteIgnoredRequest): Promise<IsRouteIgnoredResponse> {
+    throw new Error('Method not implemented.');
+  }
+  public async isRouteProtected(
+    isRouteProtectedRequest: IsRouteProtectedRequest
+  ): Promise<IsRouteProtectedResponse> {
+    throw new Error('Method not implemented.');
+  }
+  public async getDynamicOperationsByRoute(
+    getDynamicOperationsByRouteRequest: GetDynamicOperationsByRouteRequest
+  ): Promise<GetDynamicOperationsByRouteResponse> {
+    throw new Error('Method not implemented.');
+  }
+  public async getIdentityPermissionsByIdentity(
+    getIdentityPermissionsByIdentityRequest: GetIdentityPermissionsByIdentityRequest
+  ): Promise<GetIdentityPermissionsByIdentityResponse> {
+    throw new Error('Method not implemented.');
+  }
+  public async getIdentityPermissionsBySubject(
+    getIdentityPermissionsBySubjectRequest: GetIdentityPermissionsBySubjectRequest
+  ): Promise<GetIdentityPermissionsBySubjectResponse> {
+    throw new Error('Method not implemented.');
+  }
+  public async createIdentityPermissions(
+    createIdentityPermissionsRequest: CreateIdentityPermissionsRequest
+  ): Promise<CreateIdentityPermissionsResponse> {
+    const { identityPermissions } = createIdentityPermissionsRequest;
+    //Check if there are 100 ops
+    if (identityPermissions.length > 100)
+      throw new ThroughputExceededError('Exceeds 100 identity permissions');
```

Comment:
it looks like we're performing this check at the service level as well, is that necessary?

Reply:
Yes, we do not want to allow over 100+ checks to see if a group exists, that would be costly

Reply:
If the check occurs at the service, will there ever be a scenario where over 100 permissions are in the array?

Reply:
the plugin is independent of the service, still should be checked regardless. Service implementation can change